rebac-admin-backend/v1/interfaces/identity_providers.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package interfaces
+
+import (
+	"context"
+
+	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
+)
+
+// IdentityProviderService defines an abstract backend to handle Roles related operations.
+type IdentityProviderService interface {
+
+	// ListAvailableIdentityProviders returns the static list of supported identity providers.
```

Comment:
"supported" or "available"? or both have the same meaning here? I'm a bit confused with these two.

Reply:
Yes, same meaning. I prefered "available" because if they're available in this system, it's also implicit that they are supported. What is not implicit is that IdPs available for one Product, may not be available in another product